client/src/schema/repository.ts

```diff
@@ -156,7 +156,7 @@ export class RestRepository<T extends XataRecord> extends Repository<T> {
       ...fetchProps
     });
 
-    const finalObjects = await this.any(...response.recordIDs.map((id) => this.filter('id', id))).getAll();
+    const finalObjects = await this.any(...response.recordIDs.map((id) => this.filter('id' as any, id))).getAll();
```

Comment:
This actually is an error. Since id always exists it can be filtered, sorted, etc...